pkg/snet/sudpr/client.go

```diff
@@ -0,0 +1,266 @@
+package sudpr
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"net"
+	"sync"
+	"time"
+
+	"github.com/SkycoinProject/dmsg"
+	"github.com/SkycoinProject/dmsg/cipher"
+	"github.com/SkycoinProject/skycoin/src/util/logging"
+	"github.com/xtaci/kcp-go"
+
+	"github.com/SkycoinProject/skywire-mainnet/pkg/snet/arclient"
+	"github.com/SkycoinProject/skywire-mainnet/pkg/snet/directtransport"
+	"github.com/SkycoinProject/skywire-mainnet/pkg/snet/directtransport/porter"
+)
+
+// Type is sudpr type.
+const Type = "sudpr"
+
+// Client is the central control for incoming and outgoing 'sudp.Conn's.
+type Client struct {
+	log *logging.Logger
+
+	lPK             cipher.PubKey
+	lSK             cipher.SecKey
+	p               *porter.Porter
+	addressResolver arclient.APIClient
+	localAddr       string
+
+	lUDP net.Listener
+	lMap map[uint16]*directtransport.Listener // key: lPort
```

Comment:
Might be a matter of taste, but what do you think about calling this `listeners`? might be longer but not that much. it just feels to me like calling map a map is too explicit lol

Reply:
@Darkren Good idea. To not rename this in 6 clients, do you mind if I do that in #409?

Reply:
Sure